### PR TITLE
Quick CI checks for Terraform samples

### PR DESCRIPTION
**Pull request by mathieu-benoit**
Add 2 simple GitHub actions CI on any PR and commit in `main` branch for the 2 existing Terraform samples of this repo.

It's a light implementation version of https://github.com/GoogleCloudPlatform/anthos-samples/blob/main/.github/workflows/ci_any_pr.yaml.

It will allow to catch any errors raised by either `terraform init` or `terraform validate` as soon as a new PR is opened (by Renovate or any contributor for example).

### CODE REVIEW

**Comment by ckim328** [comment]
This is awesome. Thank you Mathieu for tackling this! 👏

**Comment by mathieu-benoit** [comment]
Ready for your review, thanks!

File: .github/workflows/tf-validate-ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: tf-validate-ci
+on:
+  push:
```
Comment (ckim328) [inline .github/workflows/tf-validate-ci.yml:3]: do we want it on `push` or `pr`?
https://github.com/GoogleCloudPlatform/anthos-samples/blob/main/.github/workflows/ci_any_pr.yaml#L18

Comment (ckim328) [reply .github/workflows/tf-validate-ci.yml:3]: ahh i see - would it be redundant, since we can only push onto main if the PR is up to date with main?

**Review by ckim328 — APPROVED** [review]
lgtm!

**Pull request merged**